src/foam/nanos/u2/navigation/SignIn.js

```diff
@@ -98,27 +100,32 @@ foam.CLASS({
       name: 'login',
       label: 'Sign in',
       code: async function(X) {
-        this.auth.loginByEmail(X, this.email, this.password).then(
-          (logedInUser) => {
-            if ( ! logedInUser ) return;
-            if ( this.token_ ) {
-              logedInUser.signUpToken = this.token_;
-              this.dao_.put(logedInUser)
-                .then((updatedUser) => {
-                  this.user.copyFrom(updatedUser);
-                  this.nextStep();
-                }).catch((err) => {
-                  this.notify(err.message || 'There was an issue with logging in.', 'error');
-                });
-            } else {
-              this.user.copyFrom(logedInUser);
-              this.nextStep();
+        if ( this.id.length > 0 ) {
+          this.auth.login(X, this.id, this.password).then(
+            (logedInUser) => {
+              if ( ! logedInUser ) return;
+              if ( this.token_ ) {
+                logedInUser.signUpToken = this.token_;
+                this.dao_.put(logedInUser)
+                  .then((updatedUser) => {
+                    this.user.copyFrom(updatedUser);
+                    this.nextStep();
+                  }).catch((err) => {
+                    this.notify(err.message || 'There was an issue with logging in.', 'error');
+                  });
+              } else {
+                this.user.copyFrom(logedInUser);
+                this.nextStep();
+              }
             }
-          }
-        ).catch(
-          (err) => {
-            this.notify(err.message || 'There was a problem logging in.', 'error');
-        });
+          ).catch(
+            (err) => {
+              this.notify(err.message || 'There was a problem logging in.', 'error');
+          });
+        } else {
+          //TODO: change to 'Please enter email or username' when integratting
```

Comment:
```suggestion
          //TODO: change to 'Please enter email or username' when integrating
```

and several places elsewhere. Not very important though.